src/test/java/teammates/sqlui/webapi/GetAuthInfoActionTest.java

```diff
@@ -0,0 +1,195 @@
+        JsonResult r = getJsonResult(a);
+
+        AuthInfo output = (AuthInfo) r.getOutput();
+        assertEquals(createLoginUrl("", Const.WebPageURIs.STUDENT_HOME_PAGE), output.getStudentLoginUrl());
+        assertEquals(createLoginUrl("", Const.WebPageURIs.INSTRUCTOR_HOME_PAGE), output.getInstructorLoginUrl());
+        assertEquals(createLoginUrl("", Const.WebPageURIs.ADMIN_HOME_PAGE), output.getAdminLoginUrl());
+        assertEquals(createLoginUrl("", Const.WebPageURIs.MAINTAINER_HOME_PAGE), output.getMaintainerLoginUrl());
+        assertNull(output.getUser());
+        assertFalse(output.isMasquerade());
+    }
+
+    @Test
+    void testExecute_noLoggedInUser_hasNextUrlParameter() {
+        logoutUser();
+        String nextUrl = "/web/join";
+
+        String[] params = new String[] {
+                "nextUrl", nextUrl,
+        };
+
+        GetAuthInfoAction a = getAction(params);
+        JsonResult r = getJsonResult(a);
+
+        AuthInfo output = (AuthInfo) r.getOutput();
+        assertEquals(createLoginUrl("", nextUrl), output.getStudentLoginUrl());
+        assertEquals(createLoginUrl("", nextUrl), output.getInstructorLoginUrl());
+        assertEquals(createLoginUrl("", nextUrl), output.getAdminLoginUrl());
+        assertEquals(createLoginUrl("", nextUrl), output.getMaintainerLoginUrl());
+        assertNull(output.getUser());
+        assertFalse(output.isMasquerade());
+    }
+
+    @Test
+    void testExecute_loggedInAsInstructor() {
+        loginAsInstructor("idOfInstructor1OfCourse1");
+
+        GetAuthInfoAction a = getAction();
+        JsonResult r = getJsonResult(a);
+
+        AuthInfo output = (AuthInfo) r.getOutput();
+        assertNull(output.getStudentLoginUrl());
+        assertNull(output.getInstructorLoginUrl());
+        assertNull(output.getAdminLoginUrl());
+        assertNull(output.getMaintainerLoginUrl());
+        assertFalse(output.isMasquerade());
+
+        UserInfo user = output.getUser();
+        assertFalse(user.isAdmin);
+        assertTrue(user.isInstructor);
+        assertFalse(user.isStudent);
+        assertEquals("idOfInstructor1OfCourse1", user.id);
+    }
+
+    @Test
+    void testExecute_loggedInAsUnregisteredUser() {
+        loginAsUnregistered("unregisteredId");
+
+        GetAuthInfoAction a = getAction();
+        JsonResult r = getJsonResult(a);
+
+        AuthInfo output = (AuthInfo) r.getOutput();
+        assertNull(output.getStudentLoginUrl());
+        assertNull(output.getInstructorLoginUrl());
+        assertNull(output.getAdminLoginUrl());
+        assertNull(output.getMaintainerLoginUrl());
+        assertFalse(output.isMasquerade());
+
+        UserInfo user = output.getUser();
+        assertFalse(user.isAdmin);
+        assertFalse(user.isInstructor);
+        assertFalse(user.isStudent);
+        assertEquals("unregisteredId", user.id);
+    }
+
+    @Test
+    void testExecute_addCsrfTokenCookies_noLoggedInUser() {
```

Comment:
For all of these test cases with CSRF tokens, let's keep the naming convention consistent with the other test cases (i.e. `test<functionName>_<scenario>_<outcome>`)!

If it's more convenient, I think we can also reuse the old test's method name and compile all the CSRF-related tests in the same method:

```
@Test
void testExecute_addCsrfTokenCookies_shouldAddToResponseAccordingToExistingCsrfToken() {
        ...
        
        ______TS("No logged in user");
        
        ...
}
```